benchmarks/hollow_sphere/hollow_sphere.cc

```diff
@@ -0,0 +1,704 @@
+
+        double alpha,beta,fr,gr;
+
+        if (mmm == -1)
+          {
+            alpha=-gammma*(pow(R2,3)-pow(R1,3))/(pow(R2,3)*log(R1)-pow(R1,3)*log(R2));
+            beta=-3*gammma*(log(R2)-log(R1))/(pow(R1,3)*log(R2)-pow(R2,3)*log(R1)) ;
+            fr=alpha/(r*r)+beta*r;
+            gr=-2/(r*r)*(alpha*log(r)+beta/3*pow(r,3)+gammma);
+          }
+        else
+          {
+            alpha=gammma*(mmm+1)*(pow(R1,-3)-pow(R2,-3))/(pow(R1,-mmm-4)-pow(R2,-mmm-4));
+            beta=-3*gammma*(pow(R1,mmm+1)-pow(R2,mmm+1))/(pow(R1,mmm+4)-pow(R2,mmm+4));
+            fr=alpha/pow(r,mmm+3)+beta*r;
+            gr=-2/(r*r)*(-alpha/(mmm+1)*pow(r,-mmm-1)+beta/3*pow(r,3)+gammma);
+          }
+
+        const double v_r    =gr*cos(theta);
+        const double v_theta=fr*sin(theta);
+        const double v_phi  =fr*sin(theta);
+        const double v_x=sin(theta)*cos(phi)*v_r + cos(theta)*cos(phi)*v_theta-sin(phi)*v_phi;
+        const double v_y=sin(theta)*sin(phi)*v_r + cos(theta)*sin(phi)*v_theta+cos(phi)*v_phi;
+        const double v_z=cos(theta)*v_r - sin(theta)*v_theta;
+
+        // create a Point<3> (because it has a constructor that takes
+        // three doubles) and return it (it automatically converts to
+        // the necessary Tensor<1,3>).
+        return Point<3> (v_x,v_y,v_z);
+      }
+
+      double
+      hollow_sphere_pressure (const Point<3> &pos,
+                              const double mmm)
+      {
+        //const double x = pos[0];
+        //const double y = pos[1];
+        //const double z = pos[2];
+
+        const double gammma = 1.0;
+        const double mu0=1;
+
+        const std_cxx11::array<double,3> spos =
+          aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(pos);
+
+        const double r=spos[0];
+        const double phi=spos[1];
+        const double theta=spos[2];
+
+        const double R1 = 0.5;
+        const double R2 = 1.0;
+
+        double alpha,beta,fr,gr,hr,mur;
+
+
+        if (mmm == -1)
+          {
+            mur=mu0;
+            alpha=-gammma*(pow(R2,3)-pow(R1,3))/(pow(R2,3)*log(R1)-pow(R1,3)*log(R2));
+            beta=-3*gammma*(log(R2)-log(R1))/(pow(R1,3)*log(R2)-pow(R2,3)*log(R1)) ;
+            fr=alpha/(r*r)+beta*r;
+            gr=-2/(r*r)*(alpha*log(r)+beta/3*pow(r,3)+gammma);
+            hr=2/r*gr*mur;
+          }
+        else
+          {
+            mur=mu0*pow(r,mmm+1);
+            alpha=gammma*(mmm+1)*(pow(R1,-3)-pow(R2,-3))/(pow(R1,-mmm-4)-pow(R2,-mmm-4));
+            beta=-3*gammma*(pow(R1,mmm+1)-pow(R2,mmm+1))/(pow(R1,mmm+4)-pow(R2,mmm+4));
+            fr=alpha/pow(r,mmm+3)+beta*r;
+            gr=-2/(r*r)*(-alpha/(mmm+1)*pow(r,-mmm-1)+beta/3*pow(r,3)+gammma);
+            hr=(mmm+3)/r*gr*mur;
+          }
+
+        return hr*cos(theta);
+      }
+
+
+      /**
+       * The exact solution for the Hollowsphere benchmark.
+       */
+      template <int dim>
+      class FunctionHollowsphere : public Function<dim>
```

Comment:
I think he was suggesting to call the class `HollowSphere` (with a capital `S` in the middle) to match our usual naming convention of using CamelCase for ClassNames.